### PR TITLE
style: added clang-format

### PR DESCRIPTION
**Pull request by AntimatterReactor**
**Styling and Formatting**

This PR attempts to continue #10122 (although with a somewhat different format) whilst being a lot more lax about it because it seems that formatting remains one of the more... untouched and frozen (dare I say, outdated) aspect of endless-sky.

## Summary

A `.clang-format` file is used by clang-format as a collection of option for formatting a file. Now, as of 2024 even in 19.0.0 clang-format and its options are not specific enough to adhere to the standards created by Michael Zahniser himself, mostly due to the 3, 2, and 1 blank line(s) rule. But, the good news is that clang-format is surprisingly decent.

## Testing Done

I have tried this on a few files (like AI.cpp), and to say the least clang-format is mildly successful at formatting them. There were changes, for better or worse, but it kept most of the preexisting formatting intact.

## Further Discussion

Now, here's the hard part that I can't just solve myself:

- Should clang-format be used as a regulative (every piece of code must conform) or assistive tool (just to help newer contributors and catch formatting mishaps)?
- Should we keep following a style guide that is as old as a 4th grader and still mentions SVN?
- Should I just format everything (like @Koranir did) and add it to this commit?

### CODE REVIEW

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,72 @@
+Language: Cpp
+Standard: c++17
```
Comment (TheGiraffe3) [inline .clang-format:2]: Will need to be changed when #10264 is merged.

**Comment by tibetiroka** [comment]
> I have tried this on a few files (like AI.cpp), and to say the least clang-format is mildly successful at formatting them. There were changes, for better or worse, but it kept most of the preexisting formatting intact.

We discussed similar changes internally, and there is little interest in changing our style guide to suit external tools. If clang-format breaks our formatting, then using it will make it more difficult for people to contribute, as they would have to double-check changes made by their IDE.

**Comment by gunqqer** [comment]
I'm not sure if clang-format best fits the style guide enforced, but we do need something more than the current python content style checker

Has this clang-format been ran against the entire codebase to see how much, if any, it changes?

**Comment by AntimatterReactor** [comment]
> Has this clang-format been ran against the entire codebase to see how much, if any, it changes?

Before https://github.com/endless-sky/endless-sky/pull/10306/commits/eed02b3aadac57980eb4748e01b63bca88ae3f64, it's 7307 insertions(+), 6825 deletions(-)
Now it's 5402 insertions(+), 5170 deletions(-); although I believe that number can be reduced by further tweaking

These are for source/ only btw, since I think the tests need their own formatting due to macro usages.

**Comment by TomGoodIdea** [comment]
I'd suggest closing this PR and opening a new one when it becomes possible to completely adapt clang-format to our style guide.

**Comment by AntimatterReactor** [comment]
> I'd suggest closing this PR and opening a new one when it becomes possible to completely adapt clang-format to our style guide.

Fair enough I guess

**Pull request closed without merging**